### DISCUSSION TITLE
Instance token and Class token?

### DISCUSSION THREAD

**Discussion by mykeehu** [d8ahazard/sd_dreambooth_extension#637] [General]
I would like to know what it is, what it is good for and who used it?
The quick help does not tell me anything meaningful, because if I enter the keyword I use in the Instance prompt, it extracts it, i.e. the keyword itself does not generate anything. In the past there were not these two tokens, only the Instance Prompt and Class prompt, based on these two the DB guessed the keyword. For example: 
Instance prompt: **a photo of xyz woman** 
Class prompt: **a photo of a woman** 
then the keyword became **xyz**. 
So if I typed that in itself, it would generate a picture tokened according to the sample picture. Now if I type 
Instance token: **xyz** 
Class token: **woman** 
for the tokens, it will not generate anything like what I typed in for xyz.

It's a plus that if there is a TXT with the same name and keywords next to the sample image, even if I don't specify the [filewords] tag in the Instance prompt, I think it still reads the contents. Because my prompt looks like this now: 
**keyword, prompt** 
and it does not generate what I taught it to, only if I enter one of the keywords from a txt after the prompt or without the prompt. I'm still testing it, but it seems to require a long prompt to get what I want.

So I'm getting lost in all the development, so I'm wondering what the two tokens are for, and whether or not they should be used, and how? Through examples, if someone could demonstrate, maybe I can understand better.

**Comment by mykeehu** [comment]
If [I understand this post correctly](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/443#discussioncomment-4416312), I should not use a Class prompt, but set the token like this:
- Instance token: xyz
- Class token: woman
- Instance prompt: [filewords]

In this case the woman in [filewords] is replaced by xyz, so no Class prompt is needed. Or do you have to type "woman" there too?

**Reply by mykeehu** [reply]
If I understand @d8ahazard's [comment correctly](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/667#discussioncomment-4574896), the tokens look for the two keywords in the [filewords] and complement each other. So if the 
- Instance token is: **xyz** 
- Class token is: **woman** 
then it will look for these words in the [filewords] and insert one after the other like this: **xyz woman**. 

If neither is in the [filewords], then I don't think it does anything (he didn't mention this in the comment). So in [filewords] you have to have either **xyz** or **woman** to insert the pair. I hope I understood what he wrote correctly.

**Reply by andykaufseo** [reply]
and if the tokens are empty?

**Reply by mykeehu** [reply]
I think then the classic way of keyword selection is: Class prompt - Instance prompt = keywords

**Reply by agentj01** [reply]
instance token is mostly used to call your trained dataset for testing purpose and often used for celebrities name. From what i heard you use token for training but leave it empty for fine tuning. (i fine tuned on a dataset of 6 images by just using [filewords] for class token and got great result) What really matter is the quality of images, the caption of those image and how far you overfit the traning. You can even train on 1 image, the model will still pick up on it. With the new update we have imagic to train on only 1 image, but i havent tested that yet since i am using LoRA but it is no longer working since the update, just gotta chill for a fix.

**Reply by mykeehu** [reply]
I have noticed that the more keywords I use in the class prompt (because of [filewords]), the more I will need to enter to get the right result. So I minimize the keywords here. True, this can be good for inserting new content into the given keywords, but this way you have to write more and more in the prompt, since all details have to be given.
